### ISSUE TITLE
Some ideas

### ISSUE THREAD

**Issue by GWeidmann** [chr0mcom/MethodAggregator#12]
Isn't this basically the same as the command-query pattern? Just by using methods instead of class handlers?

I like the idea, but would like to see a more generic way to setup the handler methods, e.g. by registering a whole classes methods.

And what about async methods?

**Comment by lukasschachner** [comment]
Yes you can certainly use it for implementing the command-query pattern.
Though you can use it when you dont want a direct reference to the method or dont want to inject the service/class itsself.

Registering the classes as a whole is a really nice idea.
In the past I wanted to implement it as a Attribute but just never got around to it.

Async methods could be a bit  of a headache since `void` Methods are not awaitable and so you need to be concerend about `Action` delegates.
With `Func<Task<>>` it should not have any issues.

I dont know when i will get around to implement it, if you want to push a PR i will be happy to give you a review on it

**Comment by chr0mcom** [comment]
Thanks for the idea and your opinion @GWeidmann 
Registering classes has been implemented. Only needs to be merged from @lukasschachner to Main. Next version should then be the 2.0.1.

**Comment by chr0mcom** [comment]
2.0.1 released.
async coming...
However, we will probably not implement this for a few weeks.
